### PR TITLE
Propagate proxy settings from naisd into pods

### PR DESCRIPTION
**Pull request by kimtore**
This patch allows an application to request HTTP and HTTPS proxy settings, including NO_PROXY, by defining `webproxy: true` in the nais manifest. The proxy settings are inherited from naisd's runtime environment. Both uppercase and lowercase implementations are supported, with precedence to uppercase. The default behavior is to not inject proxy settings, identical to existing behavior.

### CODE REVIEW

**Comment by ambientsound** [comment]
@audunstrand is there a difference in proxy settings for the control plane and the application, or should there be a difference? IIRC, naisd runs in the same cluster as the application?

**Comment by jhrv** [comment]
Afaik this is currently is 1:1. Gj 👌

File: api/resourcecreator.go [hunk 1]
```diff
@@ -371,22 +373,54 @@ func createEnvironmentVariables(deploymentRequest naisrequest.Deploy, naisResour
...
+				}
+				envVars = append(envVars, envVar)
+			}
+		}
+	}
+
 	return envVars, nil
 }
 
+func getenvDualCase(name string) string {
```
Comment (Kyrremann) [inline api/resourcecreator.go]: Her skulle det vel vært stor forbokstav på `env`, altså `getEnvDualCase`.

File: api/resourcecreator.go [hunk 2]
```diff
@@ -371,22 +373,54 @@ func createEnvironmentVariables(deploymentRequest naisrequest.Deploy, naisResour
 			}
 		}
 	}
+
+	// If the deployment specifies webproxy=true in the nais manifest, the pods
```
Comment (Kyrremann) [inline api/resourcecreator.go]: Usikker på om jeg er enig med at dette bør være her, men, det bør uansett også være i `nais_example.yaml`, samme med default-verdien.

Comment (ambientsound) [reply api/resourcecreator.go]: Da har jeg fikset case og dokumentert variabelen.

**Pull request merged**